.github/workflows/ci.yml

```diff
@@ -68,18 +68,38 @@ jobs:
 
   tests:
     name: Testing
-    runs-on: ${{ matrix.os }}
+    runs-on: [self-hosted, pyansys-trainings, "${{ matrix.os }}"]
     needs: [smoke-tests]
     strategy:
+      fail-fast: false
       matrix:
-          os: [ubuntu-latest, windows-latest]
           python-version: ['3.7', '3.8', '3.9', '3.10']
+          os: ['Linux', 'Windows']
     steps:
-      - name: Run pytest
-        uses: ansys/actions/tests-pytest@v4
+      - name: "Install Git and clone project"
+        uses: actions/checkout@v3
+
+      - name: "Set up Python using cache"
+        uses: actions/setup-python@v4
         with:
           python-version: ${{ matrix.python-version }}
-          pytest-extra-args: "--cov=${{ env.PACKAGE_NAMESPACE }} --cov-report=term --cov-report=xml:.cov/coverage.xml --cov-report=html:.cov/html"
+          cache: 'pip'
+          cache-dependency-path: 'pyproject.toml'
+
+      - name: "Install Linux dependencies"
+        if: runner.os == 'Linux'
+        run: |
+          sudo apt-get update
+          sudo apt-get -y install libgl1 libgl1-mesa-glx libxrender1
```

Comment:
Out of curiosity, why do we need libxrender1 here?

Reply:
Looks like otherwise it doesn't work... Seems to me like a PyVista consequence. I think we'd also need to instal opengl32.dll